stac/stac.py

```diff
@@ -79,6 +80,7 @@ def collections(self):
         :rype: dict
         """
         url = '/'.join(self._url.split('/')[:-1]) if self._url.endswith('/stac') else self._url
+        url = url[:-1] if url[-1] == '/' else url
```

Comment:
You may use `.rstrip('/')` to remove last slash entries. Or you may use [`urllib.parse.urljoin`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) for better way to deal with url path fragments join